mlrun/api/api/endpoints/artifacts_v2.py

```diff
@@ -0,0 +1,278 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from http import HTTPStatus
+from typing import List
+
+from fastapi import APIRouter, Depends, Query, Response
+from fastapi.concurrency import run_in_threadpool
+from sqlalchemy.orm import Session
+
+import mlrun.api.crud
+import mlrun.api.utils.auth.verifier
+import mlrun.api.utils.singletons.project_member
+import mlrun.common.schemas
+from mlrun.api.api import deps
+from mlrun.api.api.utils import artifact_project_and_resource_name_extractor
+from mlrun.common.schemas.artifact import ArtifactsFormat
+from mlrun.utils import logger
+
+router = APIRouter()
+
+
+@router.post("/projects/{project}/artifacts")
+async def create_artifact(
+    project: str,
+    artifact: mlrun.common.schemas.Artifact,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    await run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    key = artifact.metadata.key or None
+    tag = artifact.metadata.tag or None
+    iteration = artifact.metadata.iter or 0
+    tree = artifact.metadata.tree or None
+    logger.debug("Creating artifact", project=project, key=key, tag=tag, iter=iteration)
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.artifact,
+        project,
+        key,
+        mlrun.common.schemas.AuthorizationAction.store,
+        auth_info,
+    )
+    artifact_uid = await run_in_threadpool(
+        mlrun.api.crud.Artifacts().create_artifact,
+        db_session,
+        key,
+        artifact.dict(exclude_none=True),
+        tag,
+        iteration,
+        producer_id=tree,
+        project=project,
+    )
+    return await run_in_threadpool(
+        mlrun.api.crud.Artifacts().get_artifact,
+        db_session,
+        key,
+        tag,
+        iteration,
+        project,
+        producer_id=tree,
+        object_uid=artifact_uid,
+    )
+
+
+@router.put("/projects/{project}/artifacts/{key:path}")
+async def store_artifact(
+    project: str,
+    artifact: mlrun.common.schemas.Artifact,
+    key: str,
+    tree: str = None,
+    tag: str = None,
+    iter: int = 0,
+    object_uid: str = None,
```

Comment:
```suggestion
    object_uid: str =  Query(None, alias="object-uid"),
```
No underscore in REST params.